ddtrace/internal/processor/trace.py

```diff
@@ -68,6 +68,33 @@ def process_trace(self, trace):
         return None
 
 
+@attr.s
+class TraceTopLevelSpanProcessor(TraceProcessor):
+    """Processor marks spans as top level"""
+
+    def process_trace(self, trace):
+        # type: (List[Span]) -> Optional[List[Span]]
+        """Mark a span in a trace as top level if:
+         1. Span is a local root
+         2. Span has a different service name than its parent
+
+        Explicitly set top level to zero/false if:
+         1. Span has a truthy parent_id (not zero or None)
+            AND parent span in not in the trace chunk
```

Comment:
This explanation could be clearer: high level what is a "top level span"? What is its functional use case? What are the values and how should the be interpreted? This docstring is closer to a specification, which can certainly be helpful here to keep.

Reply:
I will add the following description: A span is "top level" if it is the first span in a trace (root span), the first span wrapping a service (span.service != parent.service), or it is the top-most span created within a process (local root). The "top level" metric will be used by the agent to calculate trace metrics and determine how spans should be displaced in the UI. If this metric is not set by the tracer the first span in a trace chunk will be marked as top level.